Add RPC inteface for getting wallet info

Traf333 opened the pull request:
### Context
this PR introduces new RPC call to provide info about signer, bitcoin and ethereum public keys



### Testing Evidences
checked only via `websocat` commands

```
echo '{"jsonrpc":"2.0","method":"bitacross_getPublicKeys","id":1}' | websocat -n1 -k -B 99999999 wss://localhost:2000

// => {"jsonrpc":"2.0","result": {"bitcoin_key":"0x023ab07017cef21e0042e123e83cebde2e838ab2bd903fb31428b45f3bd1761dfc","ethereum_key":"0x037d54c9ebee18a8b54bf3ac560015a87a16af36edc1bd8d698463cb7afb9038ab","signer":"0x953edfcc064cfa66c564739e9988233e358604baa3f0aa34c13df367661e4ffb"},"id":1}
```

Commit:
Add RPC inteface for getting wallet info

Commit:
adjust encod func

Commit:
make it look simpler

Commit:
formatter fix

Traf333 commented:
@kziemianek do we plan to have tests for RPC calls or it's not relevant at this moment?

Kailai-Wang reviewed (approved):
Looks good to me, I'm fine either one method or three. Thank you for the manual tests too.

One thing to notice is that the returned `result` is actually the SCALE-encoded `RpcReturnValue`, so extra decoding work is needed to extract the key.

kziemianek commented on bitacross-worker/enclave-runtime/src/rpc/worker_api_direct.rs:
```diff
@@ -145,6 +150,66 @@ where
+		debug!("worker_api_direct rpc was called: bitacross_getParachainKey");
+
+		async move {
+			let json_value = match GLOBAL_SIGNING_KEY_REPOSITORY_COMPONENT.get() {
```
Could you pass components to function instead of coupling to globals ?
Some of them are already passed like `top_pool_author`.

Traf333 replied:
is it ok to have functions with 7 args passing through?

kziemianek replied:
It's fine for me.

kziemianek commented on bitacross-worker/enclave-runtime/src/rpc/worker_api_direct.rs:
```diff
@@ -145,6 +150,66 @@ where
+					Ok(key) =>
+						RpcReturnValue::new(key.public().0.encode(), false, DirectRequestStatus::Ok)
+							.to_hex(),
+					Err(_) => compute_hex_encoded_return_error("Some error happened"),
```
Please log the error

kziemianek reviewed (commented):
We don't need unit tests at this point for read operations.
I left two small comments.

Traf333 commented:
> One thing to notice is that the returned result is actually the SCALE-encoded RpcReturnValue, so extra decoding work is needed to extract the key.

Do you mean, it's totally fine to return just public key without any encoding, right?

Commit:
Merge branch 'dev' into p-567-add-rpc-interface-to-retrieve-wallet-information

Commit:
simplify by passing keys through

kziemianek commented on bitacross-worker/enclave-runtime/src/rpc/worker_api_direct.rs:
```diff
@@ -74,6 +80,9 @@ pub fn public_api_rpc_handler<Author, GetterExecutor, AccessShieldingKey, OcallA
 	getter_executor: Arc<GetterExecutor>,
 	shielding_key: Arc<AccessShieldingKey>,
 	ocall_api: Arc<OcallApi>,
+	signer: ed25519::Pair,
```
I suggested passing `KeyRepository`, not keys directly. 
It's possible programaticaly to change the keys during the runtime, see:
`impl<KeyType, SealedIo> MutateKey<KeyType> for KeyRepository<KeyType, SealedIo>`

Commit:
Merge branch 'dev' into p-567-add-rpc-interface-to-retrieve-wallet-information

Commit:
pass repositories as arguments

Commit:
fmt

Commit:
turn to sync method

Commit:
turn data to hex

Commit:
Merge branch 'dev' into p-567-add-rpc-interface-to-retrieve-wallet-information

Traf333 commented:
@Kailai-Wang @kziemianek , I've modified functionality according to your comments + turned into 1 RPC method. Updated description with testing evidence

Commit:
clippy fix
